conf/route_control.py

```diff
@@ -1,530 +1,851 @@
-    else:
-        bess.resume_all()
-        print('BESS module connection ({}:{}->{}:{}) failure.'.format(
-            module, ogate, igate, next_module))
-        return
-        #raise Exception('BESS module connection ({}:{}->{}:{}) failure.'.
-        #                format(module, ogate, igate, next_module))
-
-
-def link_route_module(server, gateway_mac, item):
-    iprange = item.iprange
-    prefix_len = item.prefix_len
-    route_module = item.iface + 'Routes'
-    last_module = item.iface + 'Merge'
-    gateway_mac_str = '{:X}'.format(gateway_mac)
-    print('Adding route entry {}/{} for {}'.format(iprange, prefix_len,
-                                                   route_module))
-
-    print('Trying to retrieve neighbor entry {} from neighbor cache'.format(
-        item.neighbor_ip))
-    neighbor_exists = neighborcache.get(item.neighbor_ip)
-
-    # How many gates does this module have?
-    # If entry does not exist, then initialize it
-    if not modgatecnt.get(route_module):
-        modgatecnt[route_module] = 0
-
-    # Compute likely index
-    if neighbor_exists:
-        # No need to create a new Update module
-        gate_idx = neighbor_exists.gate_idx
-    else:
-        # Need to create a new Update module,
-        # so get gate_idx from gate count
-        gate_idx = modgatecnt[route_module]
-
-    # Pause bess first
-    bess.pause_all()
-    # Pass routing entry to bessd's route module
-    for _ in range(MAX_RETRIES):
-        try:
-            server.run_module_command(route_module, 'add',
-                                      'IPLookupCommandAddArg', {
-                                          'prefix': iprange,
-                                          'prefix_len': int(prefix_len),
-                                          'gate': gate_idx
-                                      })
-        except:
-            print('Error adding route entry {}/{} in {}. Retrying in {}sec...'.
-                  format(iprange, prefix_len, route_module, SLEEP_S))
-            time.sleep(SLEEP_S)
+            raise Exception(
+                "BESS connection failure after {} attempts.".format(self.MAX_RETRIES)
+            )
+
+    def add_route_to_module(
+        self, route_entry: RouteEntry, gate_idx: int, module_name: str
+    ) -> None:
+        """Adds a route entry to BESS.
+
+        Args:
+            route_entry (RouteEntry): Entry to be added to BESS module.
+            gate_idx (int): Gate of the module used in the route.
+            module_name (str): The name of the module.
+        """
+        for _ in range(self.MAX_RETRIES):
+            try:
+                self._bess.pause_all()
+                self._bess.run_module_command(
+                    module_name,
+                    "add",
+                    "IPLookupCommandAddArg",
+                    {
+                        "prefix": route_entry.dest_prefix,
+                        "prefix_len": int(route_entry.prefix_len),
```

Comment:
Given that `prefix_len` is already `int`, do you still need `int(route_entry.prefix_len)`?
```suggestion
                        "prefix_len": route_entry.prefix_len,
```